packages/common/src/search/wellKnownCatalog.ts

```diff
@@ -62,6 +84,51 @@ export function dotifyString(i18nScope: string): string {
   return i18nScope && i18nScope.slice(-1) !== "." ? `${i18nScope}.` : i18nScope;
 }
 
+/**
+ * Helper function to build an array of well-known Catalogs
+ * note: requested well-known Catalogs must be for the same
+ * targetEntity type
+ *
+ * @param i18nScope Translation scope to be interpolated into the catalog
+ * @param targetEntity The type of entity to query for
+ * @param catalogNames Names of the catalogs requested
+ * @param context contextual portal & auth information
+ * @param opts optional IGetWellKnownCatalogOptions args
+ */
+export function getWellKnownCatalogs(
+  i18nScope: string,
+  targetEntity: EntityType,
+  catalogNames: WellKnownCatalog[],
+  context: IArcGISContext,
+  opts?: Exclude<IGetWellKnownCatalogOptions, "context" | "user">
+): IHubCatalog[] {
+  // 1. ensure the requested Catalogs are for the same targetEntity
+  const validCatalogNames = {
+    item: [...WELL_KNOWN_ITEM_CATALOGS] as WellKnownCatalog[],
+    group: [...WELL_KNOWN_GROUP_CATALOGS] as WellKnownCatalog[],
+    event: [...WELL_KNOWN_EVENT_CATALOGS] as WellKnownCatalog[],
+  }[targetEntity as "item" | "group" | "event"];
+  if (
+    !catalogNames.every((name: WellKnownCatalog) =>
+      validCatalogNames.includes(name)
+    )
+  ) {
```

Comment:
Minor but I think it's better to use `filter` here so we can avoid duplicated checks - 

```
catalogNames.filter(
  (name: WellKnownCatalog) => !validCatalogNames.includes(name)
);
```

Reply:
Let me know if I'm misunderstanding your comment here, but the reason I'm using `every` is because I want it to throw an error if the user passes in a set of inconsistent catalog names. Using `filter` would only filter the array down to a set of consistent catalog names. Additionally, `every` will return `false` _immediately_ if any of the options don't pass the check, so I _think_ it should avoid duplicate checks.

Reply:
I see. I misunderstood it earlier. I think `every` totally works fine then! Or maybe with `some` for better readability, but it's totally your call! 

`if (catalogNames.some(name => !validCatalogNames.includes(name))) {}`